xpdacq/yamlclass.py

```diff
@@ -0,0 +1,64 @@
+##############################################################################
+#
+# xpdacq            by Billinge Group
+#                   Simon J. L. Billinge sb2896@columbia.edu
+#                   (c) 2016 trustees of Columbia University in the City
+#                   of
+#                        New York.
+#                   All rights reserved
+#
+# File coded by:    Timothy Liu
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+
+import os
+import yaml
+
+class YamlClass:
```

Comment:
I had just realized that traitlets might be able to do much of the things you were looking for this class to do. The basic idea would be that the `Glbl` class inherets from `HasTraits` then all the traits can be tracked using traitlets. Then use the `observe` functionality in traitlets to save the data down every time it changes.